### ISSUE TITLE
Request for support on OV5640 with S3.

### ISSUE THREAD

**Issue by kingmo888** [shariltumin/esp32-cam-micropython-2022#54]
Hello, I have reviewed all your documentation and found that there is currently no firmware support for the OV5640. Do you have any plans to support it?

Additionally, in other related projects, your response was that the firmware does not support the OV5640. As a beginner, I would like to ask if this means that MicroPython technically cannot support the OV5640 on the S3, or if it simply means that you do not provide support for the OV5640. Thank you.

**Comment by shariltumin** [comment]
This is correct. Currently, the compiled firmwares do not support OV5640. But, that does not mean that it will not be supported in the future version of the firmwares.I have plan to revamp the firmwares this summer with the latest MicroPython version, maybe early fall.

These have nothing to do with MicroPython, but everything to do with Espressif-IDF and esp32-camera component support. The OV5640 is supported, I just have to compile the camera driver into the firmwares.

Compiling support for OV5640 can be done. I have done it before, the problem is to get a working pin definition for OV5640 on different boards. I have had bad experiences with OV5640 camera sensors connected to ESP32-CAM boards. I "burned" a couple and gave up on the OV5640 for ESP32-CAM boards.

I am waiting for an ESP32-S3 CAM development board with OV5640 module to arrive and we will see what I can do with the latest MicroPython version with OV5640 support in the firmware.

**Comment by kingmo888** [comment]
Thank you very much for your reply. Perhaps it saved the ov5640 I just bought, but because I really didn't understand, I found the micropython amre driver project. I even wanted to simply add OV5640 to the ENUM in the. h file and try compiling it, hoping to make it run.
Fortunately, after a whole day of tinkering, the compilation failed. At this moment, I feel fortunate.

Thank you again. Looking forward to your next progress, I wish you all the best and thank you for your contribution.

**Comment by cnadler86** [comment]
If this is stil open, I would suggest to take the new API: https://github.com/cnadler86/micropython-camera-API
This should work with the 5MP cam.

**Comment by shariltumin** [comment]
Your code is so nice and clean! It's a far cry from mine, which is just a mess. I wish you all the best!

**Comment by cnadler86** [comment]
> Your code is so nice and clean! It's a far cry from mine, which is just a mess. I wish you all the best!

Hi, thanks for the positive feedback. I had a very good starting point for my code (circuitpython), so it was easy to keep it clean. 
About your code, well,  you managed to implement a functioning driver, which I will call a success 🥳, regardless of the quality of the code.
